### PR TITLE
feat: self supplied key

### PR DESCRIPTION
**Pull request by mattzcarey**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context.

## Checklist before requesting a review

Please delete options that are not relevant.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented hard-to-understand areas
- [ ] I have ideally added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged

### CODE REVIEW

File: frontend/app/upload/components/FileUploader/hooks/useFileUploader.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ export const useFileUploader = () => {
         });
       }
     },
-    [session.access_token, publish]
```
Comment (gozineb) [inline frontend/app/upload/components/FileUploader/hooks/useFileUploader.ts:41]: Do we want to remove the sesson.access_token from the dependencies ?

File: frontend/app/chat/page.tsx
Comment (gozineb) [inline frontend/app/chat/page.tsx:1]: For more readability, you could have a separate commit for these modifications -> linter only

**Comment by gozineb** [comment]
Great work, looks good to me ! ! 🚀

File: frontend/app/upload/components/FileUploader/hooks/useFileUploader.ts (hunk 1 above)
Comment (gozineb) [reply frontend/app/upload/components/FileUploader/hooks/useFileUploader.ts:41]: What linter are you using ?

Comment (mattzcarey) [reply frontend/app/upload/components/FileUploader/hooks/useFileUploader.ts:41]: I think it was how I was running the project previously not in the container. It then used my default eslint settings. It's fine now. 2 mins and ill remove that.

Comment (gozineb) [reply frontend/app/upload/components/FileUploader/hooks/useFileUploader.ts:41]: The devx is pretty bad with a container on the frontend, so you'll probably run the project outside of it again. Maybe you should change your settings or the project's to share the same

Comment (mattzcarey) [reply frontend/app/upload/components/FileUploader/hooks/useFileUploader.ts:41]: Yep agreed. I really dont like the container :)

File: frontend/app/chat/hooks/useQuestion.ts [hunk 2]
```diff
@@ -17,37 +17,44 @@ export const useQuestion = () => {
...
+
+      setHistory(response.data.history);
+      localStorage.setItem("history", JSON.stringify(response.data.history));
+      setQuestion("");
+      setIsPending(false);
+    } catch (error) {
+      console.error(error);
+    } finally {
+      setIsPending(false);
+    }
```
Comment (mamadoudicko) [inline frontend/app/chat/hooks/useQuestion.ts:51]: Since `finally` will execute in all cases, it appears that `setIsPending` is duplicated.

**Comment by mamadoudicko** [comment]
Great work !

File: frontend/app/config/hooks/useConfig.ts [hunk 3]
```diff
@@ -26,8 +27,23 @@ export const useConfig = () => {
...
+        text: "Invalid OpenAI Key",
+        variant: "danger",
+      });
+      setError("openAiKey", { type: "pattern", message: "Invalid OpenAI Key" });
+      return;
+    } else {
+      localStorage.setItem("openAiKey", values.openAiKey ?? "");
+    }
+
+    updateConfig(values);
```
[the remark this answers is not in the document]
Comment (mattzcarey) [reply frontend/app/config/hooks/useConfig.ts]: Ahh beautiful. I'll do that. I was going to ask where that went.

**Review by mamadoudicko — APPROVED** [review]

**Pull request merged**